### ISSUE TITLE
Keras example saving optimizer variables for step 0 but not step 10

### ISSUE THREAD

**Issue by rahul003** [awslabs/sagemaker-debugger#334]
```
from tests.tensorflow.keras.test_keras import train_model
train_model(
        out_dir,
        save_all=True,
        reduction_config=ReductionConfig(save_shape=True),
        use_tf_keras=True,
        save_config=SaveConfig(save_steps=[0, 10]),
        eager=False,
        steps=["train", "eval", "predict", "train"],
    )
print(create_trial_fast_refresh(out_dir).tensor_names(step=10))
verify_shapes(out_dir, 0, [])
```

**Comment by rahul003** [comment]
step 0 ['RMSprop/decay:0', 'RMSprop/iter:0', 'RMSprop/learning_rate:0', 'RMSprop/momentum:0', 'RMSprop/rho:0', 'accuracy', 'batch', 'dense/weights/dense/bias:0', 'dense/weights/dense/kernel:0', 'dense_1/weights/dense_1/bias:0', 'dense_1/weights/dense_1/kernel:0', 'loss']

step 10
['accuracy', 'batch', 'dense/weights/dense/bias:0', 'dense/weights/dense/kernel:0', 'dense_1/weights/dense_1/bias:0', 'dense_1/weights/dense_1/kernel:0', 'loss']
